### PR TITLE
vimPlugins.nvim-typescript: Init at 2020-06-06

### PR DESCRIPTION
**Pull request by bouk**
###### Motivation for this change

Add https://github.com/mhartington/nvim-typescript vim plugin

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

This is my first nix PR, I only started using it today! I have no idea what I'm doing!

I took [the code](https://github.com/NixOS/nixpkgs/pull/65472) by @nbardiuk and modified until it worked on my machine.

cc @nbardiuk  @lo1tuma

### CODE REVIEW

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -711,6 +711,35 @@ self: super: {
...
+                  --nodejs-10 \
+                  --development \
+                  --output $out/node-packages.nix \
+                  --node-env $out/node-env.nix \
+                  --composition $out/default.nix
+       ''
+     ) {});
+   in
+     super.nvim-typescript.overrideAttrs(old: {
+       buildInputs = [ nodejs ];
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix:734]: this probably isn't needed, as it won't be available at runtime
```suggestion
```

Comment (bouk) [reply pkgs/misc/vim-plugins/overrides.nix:734]: It needs nodejs to run `npm run build` though

Comment (jonringer) [reply pkgs/misc/vim-plugins/overrides.nix:734]: then it should in `nativeBuildInputs` :)

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -711,6 +711,35 @@ self: super: {
...
+         node2nix --input package.json \
+                  --lock package-lock.json \
+                  --include-peer-dependencies \
+                  --nodejs-10 \
+                  --development \
+                  --output $out/node-packages.nix \
+                  --node-env $out/node-env.nix \
+                  --composition $out/default.nix
+       ''
+     ) {});
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix:731]: this should fix your evaluation error
```suggestion
     nvim-typescript-deps = runCommand "nvim-typescript-rplugin.nix" {
         buildInputs = [ nodePackages.node2nix ];
       } ''
         mkdir -p $out
         cd ${super.nvim-typescript.src}/rplugin/node/nvim_typescript
         node2nix --input package.json \
                  --lock package-lock.json \
                  --include-peer-dependencies \
                  --nodejs-10 \
                  --development \
                  --output $out/node-packages.nix \
                  --node-env $out/node-env.nix \
                  --composition $out/default.nix
       '';
```

Comment (jonringer) [reply pkgs/misc/vim-plugins/overrides.nix:731]: actually, now that I look at this again, I'm not sure this is correct either. I'm not familiar with using node within nix to give you affirmative answer.

Comment (bouk) [reply pkgs/misc/vim-plugins/overrides.nix:731]: No, I think this is not going to work without committing a generated node2nix file.

**Comment by bouk** [comment]
I'll just close this since the approach here won't work. Thanks for the review!

**Pull request closed without merging**